integration/run_test.go

```diff
@@ -424,6 +424,36 @@ func TestRootFolderPermission(t *testing.T) {
 	}
 }
 
+func TestRestartAfterSigint(t *testing.T) {
+	const containerImage = alpineImage
+
+	regConfig := newRegistryConfig()
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	rebootContainerd(t, sh, getContainerdConfigToml(t, false), getSnapshotterConfigToml(t, false, tcpMetricsConfig))
+
+	copyImage(sh, dockerhub(containerImage), regConfig.mirror(containerImage))
+	indexDigest := buildIndex(sh, regConfig.mirror(containerImage), withMinLayerSize(0), withSpanSize(100*1024))
+	sh.X("soci", "image", "rpull", "--user", regConfig.creds(), "--soci-index-digest", indexDigest, regConfig.mirror(containerImage).ref)
+	sh.X("pkill", "-2", "soci-snapshotte") // pkill can only take up to 15 chars
+
```

Comment:
Would like some particular attention on the second half of this testing function. Mainly want to guarantee that a. the implementation would give deterministic output, b. that 5 seconds is long enough to wait for the server to crash (or not crash), and c. there are no potential race conditions (i.e. everything is guaranteed to execute before the function finishes).

Reply:
Realistically I know b and c are unlikely to be issues, but I've wrote tests in other projects that ended up being flaky due to checking for strange conditions and implemented weird workarounds, and I really don't want to do that again :v

Reply:
Is it possible that the previous snapshotter isn't fully cleaned up before the next one tries to start? Should we put a short delay in?

Reply:
I like that idea. I wonder if it's worth also doing a `ps -aux | grep soci` and printing that to log? It seems pretty unnecessary though.

Reply:
Speaking of which, maybe we could use something like `ps | grep soci` in a loop with a few hundred ms delay to determine when the process is gone rather than sleeping a full 5s. I'm not going to block on that, though.

Reply:
I like the idea, made said changes. Any chance we can apply this to the command that starts the server as well? I don't think so since lowering it would give a higher chance of a false positive, but wondering if you agree.